refactor hostkeys and knownhosts classes by moving and exposing their parameters in the relevant parents

stuartrobert opened the pull request:
- moved parameters from hostkeys to server
- removed use of knownhosts from server class
- moved parameters from knownhosts to client
- removed knownhosts class
- updated data to reflect

#### This Pull Request (PR) fixes the following issues
Fixes #411

saz commented on manifests/server.pp:
```diff
@@ -93,7 +117,17 @@
   Optional[Stdlib::Absolutepath] $sshd_environments_file = undef,
   Optional[String[1]]            $server_package_name    = undef,
+  # Host key management (used by ssh::hostkeys)
+  Boolean                        $export_ipaddresses     = true,
+  Optional[String[1]]            $storeconfigs_group     = undef,
+  Array                          $extra_aliases          = [],
+  Array                          $exclude_interfaces     = [],
+  Array                          $exclude_interfaces_re  = [],
+  Array                          $exclude_ipaddresses    = [],
+  Boolean                        $use_trusted_facts      = false,
+  Optional[Array[String[1]]]     $tags                   = undef,
```
This is a backwards-incompatible change and I don't really like it. Maybe we can just keep the parameters on `ssh::hostkeys` and use parameters on `ssh::server` as default? This way, any value set via e.g. hiera will be working as expected.

stuartrobert replied:
OK. Set it up as you described to keep backward compatibility.

saz commented on manifests/server.pp:
```diff
@@ -104,17 +138,14 @@
     -> Class['ssh::server::config']
     ~> Class['ssh::server::service']
     -> Class['ssh::hostkeys']
-    -> Class['ssh::knownhosts']
```
I think there's a chance, that someone's making use of this by not managing the client, but still making use of the collected known hosts. Except for getting rid of the class, is there any disadvantage which gets solved?

saz replied:
You're right, it's related to the client, let's stick to that and we mark this change as a breaking change.

krzee commented:
i would welcome this "breaking change". servers dont need to receive sshkeys. i found this because i started using this puppet module and came looking for how to disable this.

saz commented:
@stuartrobert can you squash commits and rebase?

stuartrobert committed:
refactor knownhosts use and allow for filtering

- moved parameters from hostkeys to server
- removed use of knownhosts from server class
- moved parameters from knownhosts to client
- removed knownhosts class
- updated data to reflect

saz committed:
update REFERENCE.md

bastelfreak commented on manifests/hostkeys.pp:
```diff
@@ -1,5 +1,5 @@
 # @summary
-#   This class manages hostkeys
+#   This class manages hostkeys. It is intended to be called from `ssh::server`.
```
Should the class be marked private with `assert_private()`?

saz commented:
Fixed with #443
